lib/redisWrapper.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+var redisWrapper = module.exports = {};
+var Q = require('q');
+var redis = require('redis');
+var _ = require('lodash');
+var redisClient;
+var separator = ':';
+var wildcard = '*';
+
+redisWrapper.initialise = function(serverConfig) {
+  var deferred = Q.defer();
+  if (_.isEmpty(serverConfig)) {
+    deferred.reject('no serverConfig passed');
+  } else {
+    redisClient = redis.createClient(serverConfig);
+    redisClient.on('ready', function () {
+      deferred.resolve();
+    });
+    // if there was error connecting to redis server
+    redisClient.on('error', function () {
+      deferred.reject( 'Error trying to connect to redis server');
+    });
+
+  }
+  return deferred.promise;
+};
+/**
+* Allows you to retrieve a keys which have a value matching the pattern passed in prefix.
+* Using SCAN here and not KEYS to avoid blocking the server for a long time when called against big collections of keys or elements.
```

Comment:
Really good find here, *SCAN* over *KEYS* is a great performance gain and something that could have caused us issues in production. Great stuff.